assets/js/components/dashboard-sharing/ModuleRecoveryAlert/index.js

```diff
@@ -19,251 +19,104 @@
 /**
  * WordPress dependencies
  */
-import { Fragment, useCallback, useEffect, useState } from '@wordpress/element';
-import { sprintf, __ } from '@wordpress/i18n';
+import { __ } from '@wordpress/i18n';
 
 /**
  * Internal dependencies
  */
-import { useSelect, useDispatch } from 'googlesitekit-data';
-import { Checkbox, ProgressBar } from 'googlesitekit-components';
-import { CORE_SITE } from '../../../googlesitekit/datastore/site/constants';
+import { useSelect } from 'googlesitekit-data';
 import { CORE_MODULES } from '../../../googlesitekit/modules/datastore/constants';
-import { DAY_IN_SECONDS } from '../../../util';
-import BannerNotification from '../../notifications/BannerNotification';
-import Errors from './Errors';
-
-export default function ModuleRecoveryAlert() {
-	const [ checkboxes, setCheckboxes ] = useState( null );
-	const [ recoveringModules, setRecoveringModules ] = useState( false );
+import SimpleNotification from '../../../googlesitekit/notifications/components/layout/SimpleNotification';
+import Description from './Description';
+import RecoverableActions from './RecoverableActions';
+import UnrecoverableActions from './UnrecoverableActions';
 
+export default function ModuleRecoveryAlert( { id, Notification } ) {
 	const recoverableModules = useSelect( ( select ) =>
 		select( CORE_MODULES ).getRecoverableModules()
 	);
 
-	const documentationURL = useSelect( ( select ) => {
-		return select( CORE_SITE ).getDocumentationLinkURL(
-			'dashboard-sharing'
-		);
-	} );
-
-	const userAccessibleModules = useSelect( ( select ) => {
-		const modules = select( CORE_MODULES ).getRecoverableModules();
+	// TODO: Extract to selector.
+	const userRecoverableModuleSlugs = useSelect( ( select ) => {
+		const { getRecoverableModules, hasModuleAccess } =
+			select( CORE_MODULES );
+		const modules = getRecoverableModules();
 
 		if ( modules === undefined ) {
 			return undefined;
 		}
 
-		const accessibleModules = Object.keys( modules ).map( ( slug ) => ( {
+		const slugAccessEntries = Object.keys( modules ).map( ( slug ) => [
```

Comment:
More accurately named: this is a collection of slugs with their "has access" value, but the previous name implied they were all accessible.